### PR TITLE
Identity Deletion Job: Add Docker support

### PR DESCRIPTION
**Pull request by MH321Productions**
# Readiness checklist

-   [ ] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

**Comment by MH321Productions** [comment]
Note: In Order for the container to find RabbitMQ and Postgres, it has to be in the backbone network

File: Jobs/src/Job.IdentityDeletion/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+COPY ["BuildingBlocks/src/BuildingBlocks.Application/BuildingBlocks.Application.csproj", "BuildingBlocks/src/BuildingBlocks.Application/"]
+COPY ["Common/src/Common.Infrastructure/Common.Infrastructure.csproj", "Common/src/Common.Infrastructure/"]
+COPY ["BuildingBlocks/src/BuildingBlocks.Infrastructure/BuildingBlocks.Infrastructure.csproj", "BuildingBlocks/src/BuildingBlocks.Infrastructure/"]
+COPY ["BuildingBlocks/src/DevelopmentKit.Identity/DevelopmentKit.Identity.csproj", "BuildingBlocks/src/DevelopmentKit.Identity/"]
+COPY ["BuildingBlocks/src/BuildingBlocks.Domain/BuildingBlocks.Domain.csproj", "BuildingBlocks/src/BuildingBlocks.Domain/"]
+COPY ["BuildingBlocks/src/Tooling/Tooling.csproj", "BuildingBlocks/src/Tooling/"]
+COPY ["BuildingBlocks/src/BuildingBlocks.Application.Abstractions/BuildingBlocks.Application.Abstractions.csproj", "BuildingBlocks/src/BuildingBlocks.Application.Abstractions/"]
+COPY ["Infrastructure/Infrastructure.csproj", "Infrastructure/"]
+
+RUN dotnet restore -r linux-x64 "Jobs/src/Job.IdentityDeletion/Job.IdentityDeletion.csproj"
```
Comment (tnotheis) [inline Jobs/src/Job.IdentityDeletion/Dockerfile:26]: Why do you specify `-r linux-x64` explicitly? (same for the other dotnet commands in this Dockerfile)

File: Jobs/src/Job.IdentityDeletion/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+COPY . .
+
+WORKDIR "/src/Jobs/src/Job.IdentityDeletion"
+RUN dotnet build "Job.IdentityDeletion.csproj" --configuration Release -r linux-x64 --output /app/build --no-restore
+
+FROM build AS publish
+RUN dotnet publish -c Release --output /app/publish -r linux-x64 --self-contained true --no-restore "Job.IdentityDeletion.csproj"
```
Comment (tnotheis) [inline Jobs/src/Job.IdentityDeletion/Dockerfile:34]: Can you try to make it run without the `--self-contained`? @NikolaVetnic  told me that he was able to make this work. Maybe you can use his solution.

The problem with self-contained is that the binaries (and therefore the size of the Docker image) are much larger.

(I know that I suggested using self-contained, but tbh I don't know why I said that. ;)

File: Jobs/src/Job.IdentityDeletion/appsettings.override.json
Comment (tnotheis) [inline Jobs/src/Job.IdentityDeletion/appsettings.override.json:1]: You shouldn't commit the changes in this file. If we run the job locally, we do it outside of Docker, so we need "localhost"

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Jobs/src/Job.IdentityDeletion/Dockerfile (hunk 1 above)
Comment (NikolaVetnic) [reply Jobs/src/Job.IdentityDeletion/Dockerfile:26]: The `-r linux-x64` part can be removed from `dotnet build` and `dotnet publish` commands, however it has to remain in the `dotnet restore`. Removing it in that place as well causes this error:
```shell
 > [publish 1/1] RUN dotnet publish -c Release --output /app/publish --self-contained true --no-restore "Job.IdentityDeletion.csproj":
0.831 MSBuild version 17.9.4+90725d08d for .NET
1.660 /usr/share/dotnet/sdk/8.0.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): error NETSDK1047: Assets file '/src/Jobs/src/Job.IdentityDeletion/obj/project.assets.json' doesn't have a target for 'net8.0/linux-x64'. Ensure that restore has run and that you have included 'net8.0' in the TargetFrameworks for your project. You may also need to include 'linux-x64' in your project's RuntimeIdentifiers. [/src/Jobs/src/Job.IdentityDeletion/Job.IdentityDeletion.csproj]
```

File: Jobs/src/Job.IdentityDeletion/Dockerfile (hunk 2 above)
Comment (NikolaVetnic) [reply Jobs/src/Job.IdentityDeletion/Dockerfile:34]: Changes made.

File: Jobs/src/Job.IdentityDeletion/appsettings.override.json
Comment (NikolaVetnic) [reply Jobs/src/Job.IdentityDeletion/appsettings.override.json:1]: I've reverted the changes made to this file.

File: Jobs/src/Job.IdentityDeletion/Job.IdentityDeletion.csproj [hunk 3]
```diff
@@ -1,9 +1,11 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Worker">
 
     <PropertyGroup>
-        <UserSecretsId>dotnet-Devices.Jobs.IdentityDeletion-553687ad-d9d2-47b9-88c5-a8869b426c22</UserSecretsId>
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
```
Comment (tnotheis) [inline Jobs/src/Job.IdentityDeletion/Job.IdentityDeletion.csproj:6]: These can and should be removed. We want to control this via our `Directory.Build.props` files.

File: Jobs/src/Job.IdentityDeletion/Program.cs [hunk 4]
```diff
@@ -97,6 +98,11 @@ static IHostBuilder CreateHostBuilder(string[] args)
                 .AddModule<SynchronizationModule>(configuration)
                 .AddModule<TokensModule>(configuration);
 
+            services.AddMediatR(c => c
+                .RegisterServicesFromAssemblyContaining<CancelStaleIdentityDeletionProcessesCommand>());
```
Comment (tnotheis) [inline Jobs/src/Job.IdentityDeletion/Program.cs:102]: Can be removed. This is called in `AddModule<DevicesModule>`

File: Jobs/src/Job.IdentityDeletion/Program.cs [hunk 5]
```diff
@@ -97,6 +98,11 @@ static IHostBuilder CreateHostBuilder(string[] args)
...
+
+            services.AddModule<DevicesModule>(configuration);
```
Comment (tnotheis) [inline Jobs/src/Job.IdentityDeletion/Program.cs:104]: Can be removed. The DevicesModule is already added in line 92.

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Jobs/src/Job.IdentityDeletion/Job.IdentityDeletion.csproj (hunk 3 above)
Comment (NikolaVetnic) [reply Jobs/src/Job.IdentityDeletion/Job.IdentityDeletion.csproj:6]: Changes made.

File: Jobs/src/Job.IdentityDeletion/Program.cs (hunk 4 above)
Comment (NikolaVetnic) [reply Jobs/src/Job.IdentityDeletion/Program.cs:102]: Changes made.

File: Jobs/src/Job.IdentityDeletion/Program.cs (hunk 5 above)
Comment (NikolaVetnic) [reply Jobs/src/Job.IdentityDeletion/Program.cs:104]: Changes made.

**Review by tnotheis — APPROVED** [review]

**Pull request merged**